langwatch_nlp/tests/test_secret_interpolation.py

```
@@ -0,0 +1,70 @@
+            assert "KEY_B='val_b'" in preamble
+
+    class TestWhenSecretsContainSpecialCharacters:
+        def test_escapes_quotes_via_repr(self):
+            preamble = build_secrets_preamble(
+                {"TRICKY": 'has "double" and \'single\' quotes'}
+            )
+            # repr() will properly escape quotes
+            assert "TRICKY=" in preamble
+            # The value should be a valid Python literal (repr handles escaping)
+            exec_globals: dict = {}
+            exec(preamble, exec_globals)
```

Ruff (0.15.1)

[error] Use of `exec` detected

(S102)